vmm/src/acpi.rs

```diff
@@ -197,6 +197,34 @@ pub fn create_acpi_tables(
                 });
             }
 
+            for region in node.hotplug_regions() {
+                let base_addr = region.start_addr().raw_value();
+                let base_addr_lo = (base_addr & 0xffff_ffff) as u32;
+                let base_addr_hi = (base_addr >> 32) as u32;
+                let length = region.len() as u64;
+                let length_lo = (length & 0xffff_ffff) as u32;
+                let length_hi = (length >> 32) as u32;
+
+                // Flags
+                // - Enabled = 1 (bit 0)
+                // - Hot Pluggable = 0 (bit 1)
+                // - NonVolatile = 0 (bit 2)
+                // - Reserved bits 3-31
+                let flags = 3;
+
+                srat.append(MemoryAffinity {
+                    type_: 1,
+                    length: 40,
+                    proximity_domain,
+                    base_addr_lo,
+                    base_addr_hi,
+                    length_lo,
+                    length_hi,
+                    flags,
+                    ..Default::default()
+                });
```

Comment:
Could you please factorize the `MemoryAffinity` structure creation into a dedicated function, so that both `memory_regions` and `hotplug_regions` can reuse it. The difference between the two is simply about the flags being differents, so I think you can come up with a function like that:
```rust
impl MemoryAffinity {
    fn from_region(region: Arc<GuestRegionMmap>, flags: u32) -> Self {
        ...
    }
}
```

Reply:
Done, please have a review. :)